app/docker/helpers/containerHelper.js

```diff
@@ -54,5 +54,170 @@ angular.module('portainer.docker')
     return config;
   };
 
+  helper.parsePortRange = function(ports) {
```

Comment:
* This function can be refactored to use the following regex
```
const portPattern = /^()([1-9]|[1-5]?[0-9]{2,4}|6[1-4][0-9]{3}|65[1-4][0-9]{2}|655[1-2][0-9]|6553[1-5])$/gm;
```
This ensures that the port is in 1-65535 range.

* Always return `[start, end]` (end = start when there is a single port)

Reply:
The regex pattern that you provided fails to recognize ports such as 60000, 65000, 65500, etc.
I made some minor adjustments to resolve this:
`^([1-9]|[1-5]?[0-9]{2,4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$`